worlds/zork_grand_inquisitor/world.py

```diff
@@ -0,0 +1,210 @@
+    Foundation, and The Skull of Yoruk.
+    """
+
+    options_dataclass = ZorkGrandInquisitorOptions
+    options: ZorkGrandInquisitorOptions
+
+    game = "Zork Grand Inquisitor"
+
+    topology_present = False
+
+    item_name_to_id = item_names_to_id()
+    location_name_to_id = location_names_to_id()
+
+    item_name_groups = item_groups()
+    location_name_groups = location_groups()
+
+    required_client_version: Tuple[int, int, int] = (0, 4, 4)
+
+    web = ZorkGrandInquisitorWebWorld()
+
+    item_name_to_item: Dict[str, ZorkGrandInquisitorItems] = item_names_to_item()
+
+    def create_regions(self) -> None:
+        deathsanity: bool = self.options.deathsanity.value == 1
+
+        region_mapping: Dict[ZorkGrandInquisitorRegions, Region] = dict()
+
+        region_enum_item: ZorkGrandInquisitorRegions
+        for region_enum_item in region_data.keys():
+            region_mapping[region_enum_item] = Region(region_enum_item.value, self.player, self.multiworld)
+
+        region_locations_mapping: Dict[ZorkGrandInquisitorRegions, Set[ZorkGrandInquisitorLocations]]
+        region_locations_mapping = locations_by_region(include_deathsanity=deathsanity)
+
+        region_enum_item: ZorkGrandInquisitorRegions
+        region: Region
+        for region_enum_item, region in region_mapping.items():
+            regions_locations: Set[ZorkGrandInquisitorLocations] = region_locations_mapping[region_enum_item]
+
+            # Locations
+            location_enum_item: ZorkGrandInquisitorLocations
+            for location_enum_item in regions_locations:
+                data: ZorkGrandInquisitorLocationData = location_data[location_enum_item]
+
+                location: ZorkGrandInquisitorLocation = ZorkGrandInquisitorLocation(
+                    self.player,
+                    location_enum_item.value,
+                    data.archipelago_id,
+                    region_mapping[data.region],
+                )
+
+                location.event = isinstance(location_enum_item, ZorkGrandInquisitorEvents)
+
+                if location.event:
+                    location.place_locked_item(
+                        ZorkGrandInquisitorItem(
+                            data.event_item_name,
+                            ItemClassification.progression,
+                            None,
+                            self.player,
+                        )
+                    )
+
+                location_access_rule: str = location_access_rule_for(location_enum_item, self.player)
+
+                if location_access_rule != "lambda state: True":
+                    location.access_rule = eval(location_access_rule)
+
+                region.locations.append(location)
+
+            # Connections
+            region_exit: ZorkGrandInquisitorRegions
+            for region_exit in region_data[region_enum_item].exits or tuple():
```

Comment:
what's the point of the tuple here?

Reply:
`exits` can be None. I prefer to have None in my data vs an empty tuple when there are no exits (good old null not being the same as 0)